brave-intl/publishers
# Manual Payout Report 🏷

![screen shot 2019-03-06 at 10 48 35 pm](https://user-images.githubusercontent.com/4713771/53931149-04063880-4062-11e9-989a-3e2beef30bef.png)

Closes #1603 

#### Features
- New `Manual` type for payout reports, which contains partners that have finalized invoices. 
- Optionally generate a `manual` payout report for a single selected `partner`. 

#### Notes
- Regular payout reports will not include manual entries, they will be kept distinct. 
- Manual payout report `amount` will be the `finalized amount` set by BizDev
- Manual payout reports on publishers will not be generated monthly, although it is easy to generate them at any time. 

#### How to Test
- As an admin, request an invoice from a partner
- Partner may upload supporting documents
- As an admin, finalize the partner's invoice
- After finalizing, you will see a `generate manual payout` button, which will generate an single entry
- Alternatively admin can generate a `manual` type payout from the `payout reports` tab, for all Partners

#### Note

We need to make sure to nil out the invoice once the partner is paid so it is not included in the next report. 

#### TODO 

- [x] Wire in Generate Manual Report button
- [x] Ensure the rest of the payout report works as intended
- [x] Ensure only partners with finalized invoice are included 
- [x] Move manual payout json to its own file
- [x] Ensure finalized amount to probi handles floats/commas correctly
